packages/selected-network-controller/src/SelectedNetworkMiddleware.ts

```diff
@@ -37,28 +37,11 @@ export const createSelectedNetworkMiddleware = (
       origin,
     );
 
-  const setNetworkClientIdForDomain = (
-    origin: string,
-    networkClientId: NetworkClientId,
-  ) =>
-    messenger.call(
-      SelectedNetworkControllerActionTypes.setNetworkClientIdForDomain,
-      origin,
-      networkClientId,
-    );
-
-  const getDefaultNetworkClientId = () =>
-    messenger.call('NetworkController:getState').selectedNetworkClientId;
```

Comment:
With the removal of both `NetworkController:{getState,stateChange}` from this file, the `MiddlewareMessenger` pattern is no longer useful. We could replace it with `SelectedNetworkControllerMessenger` and lose some unnecessary bloat. 

This could be done in its own PR, but feel free to apply this diff here if it's helpful.

```diff
diff --git a/packages/selected-network-controller/src/SelectedNetworkMiddleware.ts b/packages/selected-network-controller/src/SelectedNetworkMiddleware.ts
index eea09733..eb84a503 100644
--- a/packages/selected-network-controller/src/SelectedNetworkMiddleware.ts
+++ b/packages/selected-network-controller/src/SelectedNetworkMiddleware.ts
@@ -1,35 +1,17 @@
-import type { ControllerMessenger } from '@metamask/base-controller';
 import type { JsonRpcMiddleware } from '@metamask/json-rpc-engine';
-import type {
-  NetworkClientId,
-  NetworkControllerGetStateAction,
-  NetworkControllerStateChangeEvent,
-} from '@metamask/network-controller';
+import type { NetworkClientId } from '@metamask/network-controller';
 import type { Json, JsonRpcParams, JsonRpcRequest } from '@metamask/utils';
 
-import type {
-  SelectedNetworkControllerGetNetworkClientIdForDomainAction,
-  SelectedNetworkControllerSetNetworkClientIdForDomainAction,
-} from './SelectedNetworkController';
+import type { SelectedNetworkControllerMessenger } from './SelectedNetworkController';
 import { SelectedNetworkControllerActionTypes } from './SelectedNetworkController';
 
-export type MiddlewareAllowedActions = NetworkControllerGetStateAction;
-export type MiddlewareAllowedEvents = NetworkControllerStateChangeEvent;
-
-export type SelectedNetworkMiddlewareMessenger = ControllerMessenger<
-  | SelectedNetworkControllerGetNetworkClientIdForDomainAction
-  | SelectedNetworkControllerSetNetworkClientIdForDomainAction
-  | MiddlewareAllowedActions,
-  MiddlewareAllowedEvents
->;
-
 export type SelectedNetworkMiddlewareJsonRpcRequest = JsonRpcRequest & {
   networkClientId?: NetworkClientId;
   origin?: string;
 };
 
 export const createSelectedNetworkMiddleware = (
-  messenger: SelectedNetworkMiddlewareMessenger,
+  messenger: SelectedNetworkControllerMessenger,
 ): JsonRpcMiddleware<JsonRpcParams, Json> => {
   const getNetworkClientIdForDomain = (origin: string) =>
     messenger.call(
```

Reply:
done here: https://github.com/MetaMask/core/pull/3908/commits/3cd255439e02278dd8381e2bb6ae53a7348041f5